pilot/pkg/networking/deprecated/listener.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deprecated
+
+import (
+	"fmt"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/envoy/api/v2/listener"
+	"github.com/gogo/protobuf/types"
+
+	"strings"
+
+	"github.com/kylelemons/godebug/pretty"
+
+	google_protobuf "github.com/gogo/protobuf/types"
+	_ "github.com/golang/glog" // nolint
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+)
+
+// LdsConnection is a listener connection type.
+type LdsConnection struct {
```

Comment:
this connection stuff belongs in proxy/envoy/v2/lds.go ..

Reply:
Did you see the recent interface.go that went into networking/ ? 
LdsDiscoveryResponse seems to only need BuildListeners.. and you could do it cleanly using
```
networkConfigGen := networking.NewNetworking()
ls, err := networkConfigGen.BuildListeners(env, node) ...
```

This will cleanly isolate lds connection or whatever from the networking code, wont it?

Reply:
So that's option 1, which I'm perfectly ok with. But since @costinm was arguing for option 3 I'll let him respond.

Reply:
I'm happy any way that gets this PR in and e2e tested. I was very happy to have all the files in one place (v2) and not debating which of the 3 places to place stuff. If Martin put it in there, I'll trust him and let someone else move them to better places if needed.

Reply:
Ok, I've moved it back to v2, I think that's a reasonable place to keep it since it has more to do with connections than with the building of the response.